src/main/java/org/bankofspring/dao/jdbc/BankOfSpringDAOImpl.java

```diff
@@ -0,0 +1,183 @@
+	private final String ACCOUNT_UPDATE = "UPDATE account set balance=? where number = ?";
+	
+	private SimpleJdbcTemplate simpleTemplate;
+	private NamedParameterJdbcTemplate namedTemplate;
+	
+	@Autowired
+	public void setDataSource(DataSource source) {
+		this.simpleTemplate = new SimpleJdbcTemplate(source);
+		this.namedTemplate = new NamedParameterJdbcTemplate(source);
+	}
+	
+	public SimpleJdbcTemplate getSimpleTemplate() {
+		return simpleTemplate;
+	}
+	
+	public NamedParameterJdbcTemplate getNamedTemplate() {
+		return namedTemplate;
+	}
+	
+	@Override
+	public List<Customer> getCustomers() {
+		 return getSimpleTemplate().query(CUSTOMERS_SELECT, new RowMapper<Customer>() {
+
+			@Override
+			public Customer mapRow(ResultSet res, int rowNum)
+					throws SQLException {
+				String id = res.getString(1);
+				String name = res.getString(2);
+				String userName = res.getString(3);
+				String password = res.getString(4);
+				Customer customer = new Customer(userName,password,name,id);
+				return customer;
+				
+			}
+			
+		});
+	}
+
+	@Override
+	public List<Customer> getCustomersForAccount(String accountNumber) {
+		 return getSimpleTemplate().query(CUSTOMERS_FOR_ACCOUNT_SELECT, new RowMapper<Customer>() {
+
+			@Override
+			public Customer mapRow(ResultSet res, int rowNum)
+					throws SQLException {
+				String id = res.getString(1);
+				String name = res.getString(2);
+				String userName = res.getString(3);
+				String password = res.getString(4);
+				Customer customer = new Customer(userName,password,name,id);
+				return customer;
+				
+			}
+			
+		}, accountNumber);
+	}
+
+	@Override
+	public Account getAccount(String accountNumber) {
+		return getSimpleTemplate().queryForObject(ACCOUNT_SELECT, new RowMapper<Account>() {
+		
+			@Override
+			public Account mapRow(ResultSet res, int rowNum)
+					throws SQLException {
+				String id = res.getString(1);
+				String description = res.getString(2);
+				long balance = res.getLong(3);
+				Account account = new Account(id,description,null);
+				account.setAccountBalance(balance);
+				List<Customer> customers = getCustomersForAccount(account.getAccountNumber());
+				account.setOwningCustomers(customers);
+				return account;
+				
+			}
+			
+		},accountNumber);
+	}
+
+
+	@Override
+	public Customer getCustomer(String customerId) {
+		return getSimpleTemplate().queryForObject(SINGLE_CUSTOMER_SELECT, new RowMapper<Customer>() {
+			
+			@Override
+			public Customer mapRow(ResultSet res, int rowNum)
+					throws SQLException {
+				String id = res.getString(1);
+				String name = res.getString(2);
+				String userName = res.getString(3);
+				String password = res.getString(4);
+				Customer customer = new Customer(userName,password,name,id);
+				return customer;
+			}
+		},customerId);
+	}
+
+
+	@Override
+	public Account updateAccountBalance(String accountNumber, long balance) {
+		 getSimpleTemplate().update(ACCOUNT_UPDATE, balance,accountNumber);
```

Comment:
What happens if the balance changes between your validation and this query running?